FirebaseStorageSwift/Tests/Unit/StorageComponentTests.swift

```diff
@@ -0,0 +1,92 @@
+// limitations under the License.
+
+import Foundation
+
+import FirebaseCore
+@testable import FirebaseStorage
+
+import FirebaseAppCheckInterop
+import FirebaseAuthInterop
+import SharedTestUtilities
+
+import XCTest
+
+class StorageComponentTests: XCTestCase {
+  static var app: FirebaseApp?
+
+  override class func setUp() {
+    let options = FirebaseOptions(googleAppID: "0:0000000000000:ios:0000000000000000",
+                                  gcmSenderID: "00000000000000000-00000000000-000000000")
+    options.projectID = "myProjectID"
+    FirebaseApp.configure(name: "test", options: options)
+    app = FirebaseApp(instanceWithName: "test", options: options)
+  }
+
+  // MARK: Interoperability Tests
+
+  /// Tests that the right number of components are being provided for the container.
+  func testComponentsBeingRegistered() throws {
+    let components = StorageComponent.componentsToRegister()
+    XCTAssert(components.count == 1)
+  }
+
+  /// Tests that the right number of components are being provided for the container.
+  func testStorageInstanceCreation() throws {
+    let component = StorageComponent(app: StorageComponentTests.app!)
+    let storage = component.storage(for: "someBucket")
+    XCTAssertNotNil(storage)
+  }
+
+  /// Tests that the component container caches instances of StorageComponent.
+  func testMultipleComponentInstancesCreated() throws {
+    let app = try XCTUnwrap(StorageComponentTests.app)
+    let registrants = NSMutableSet(array: [StorageComponent.self])
+    let container = FirebaseComponentContainer(app: app, registrants: registrants)
+
+    let provider1 = ComponentType<StorageProvider>.instance(for: StorageProvider.self,
+                                                            in: container)
+    XCTAssertNotNil(provider1)
+
+    let provider2 = ComponentType<StorageProvider>.instance(for: StorageProvider.self,
+                                                            in: container)
+    XCTAssertNotNil(provider2)
+
+    // Ensure they're the same instance.
+    XCTAssert(provider1 === provider2)
+  }
+
+  /// Tests that instances of Storage created are different.
+  func testMultipleStorageInstancesCreated() throws {
+    let app = try XCTUnwrap(StorageComponentTests.app)
+    let registrants = NSMutableSet(array: [StorageComponent.self])
+    let container = FirebaseComponentContainer(app: app, registrants: registrants)
+
+    let provider = ComponentType<StorageProvider>.instance(for: StorageProvider.self,
+                                                           in: container)
+    XCTAssertNotNil(provider)
+
+    let storage1 = provider.storage(for: "randomBucket")
+    let storage2 = provider.storage(for: "randomBucket")
+    XCTAssertNotNil(storage1)
+
+    // Ensure they're the same instance.
+    XCTAssert(storage1 === storage2)
+
+    let storage3 = provider.storage(for: "differentBucket")
+    XCTAssertNotNil(storage3)
+
+    XCTAssertNotEqual(storage1, storage3)
```

Comment:
Hmm I'm wondering why we can use `NotEqual` here but can't use `Equal` above... does this compare instances under the hood? Or should we use `!==`?